File: Python/practical/53.py
# Write a program to accept two matrices and subtract them

def matrixInput(size):
    matrix = []
    submatrix = []
    
    for i in range(size):
        for j in range(size):
            element = int(input(f"Enter Element ({i}, {j}): "))
            submatrix.append(element)
        
        matrix.append(submatrix)
        submatrix = []
    
    return matrix

def matrixMul(size, matrixA, matrixB):
    resMatrix1 = [[0 for i in range(size)] for j in range(size)]
    resMatrix2 = [[0 for i in range(size)] for j in range(size)]
    
    for i in range(size):
        for j in range(size):
            for k in range(size):
                resMatrix1[i][j] = resMatrix1[i][j] + (matrixA[i][k] * matrixB[k][j])
                resMatrix2[i][j] = resMatrix2[i][j] + (matrixB[i][k] * matrixA[k][j])
    
    return resMatrix1, resMatrix2

size = int(input("Enter the size of the matrices: "))

print("Matrix A: ")
matrixA = matrixInput(size)

print("Matrix B: ")
matrixB = matrixInput(size)

resMatrix1, resMatrix2 = matrixMul(size, matrixA, matrixB)

print ("A * B = ")

for i in range(size):
    for j in range(size):
        print(f"{resMatrix1[i][j]}\t", end = '')
    
    print()

print ("B * A = ")

for i in range(size):
    for j in range(size):
        print(f"{resMatrix2[i][j]}\t", end = '')
    
    print()

File: C/SEM I/Practical/Done/53.c
/* Write a program to find out the product/Multiplication of two matrices and print the product matrix.(order of matrices must be given by user) */

#include <stdio.h>
#include <conio.h>
#include <stdlib.h>

typedef int intArr;
typedef int loopVal;

int count = 0;

int main(void) {
	int rep = 1;

	while (rep == 1) {
		intArr matrixA[50][50], matrixB[50][50], matrixC[50][50];
		loopVal i, j, k;
		int rowSizeA, colSizeA;
		int rowSizeB, colSizeB;

		printf("Enter matrix A row size: ");
		scanf("%d", & rowSizeA);
		printf("Enter matrix A column size: ");
		scanf("%d", & colSizeA);

		printf("Enter matrix B row size: ");
		scanf("%d", & rowSizeB);
		printf("Enter matrix B column size: ");
		scanf("%d", & colSizeB);

		if (rowSizeA == colSizeB) {
			for (i = 0; i < 50; i++) {
				for (j = 0; j < 50; j++) {
					matrixA[i][j] = 0;
					matrixB[i][j] = 0;
					matrixC[i][j] = 0;
				}
			}

			printf("Enter matrix A: \n");

			for (i = 0; i < rowSizeA; i++) {
				for (j = 0; j < colSizeA; j++) {
					printf("Enter element (%d, %d): ", i, j);
					scanf("%d", & matrixA[i][j]);
				}
			}

			printf("Enter matrix B: \n");

			for (i = 0; i < rowSizeB; i++) {
				for (j = 0; j < colSizeB; j++) {
					printf("Enter element (%d, %d): ", i, j);
					scanf("%d", & matrixB[i][j]);
				}
			}

			for (i = 0; i < 50; i++) {
				for (j = 0; j < 50; j++) {
					for (k = 0; k < 50; k++) {
						matrixC[i][j] += matrixA[i][k] * matrixB[k][j];
					}
				}
			}

			printf("\nMatrix A: \n");

			for (i = 0; i < rowSizeA; i++) {
				for (j = 0; j < colSizeA; j++) {
					printf("%d\t", matrixA[i][j]);
				}

				printf("\n");
			}

			printf("\nMatrix B: \n");

			for (i = 0; i < rowSizeB; i++) {
				for (j = 0; j < colSizeB; j++) {
					printf("%d\t", matrixB[i][j]);
				}

				printf("\n");
			}

			printf("Resultant matrix C is: \n");

			for (i = 0; i < rowSizeA; i++) {
				for (j = 0; j < colSizeB; j++) {
					printf("%d\t", matrixC[i][j]);
				}

				printf("\n");
			}

		} else {
			printf("Matrix multplication is not possible for given matrices.");
		}

		printf("\nRepeat Program?: ");
		scanf("%d", & rep);

		count++;

		if (count >= 50) {
			printf("Abnormal END\n");

			getch();
			exit(0);
		}

	}

	printf("Normal END\n");

	getch();
	return 0;
}